mailpoet/lib/Automation/Integrations/MailPoet/Actions/SendEmailAction.php

```diff
@@ -128,17 +135,26 @@ public function run(StepRunArgs $args): void {
     }
 
     $subscriberStatus = $subscriber->getStatus();
-    if ($subscriberStatus !== SubscriberEntity::STATUS_SUBSCRIBED) {
+    if (!$this->isTransactional($args) && $subscriberStatus !== SubscriberEntity::STATUS_SUBSCRIBED) {
       throw InvalidStateException::create()->withMessage(sprintf("Cannot schedule a newsletter for subscriber ID '%s' because their status is '%s'.", $subscriberId, $subscriberStatus));
     }
 
     try {
-      $this->automationEmailScheduler->createSendingTask($newsletter, $subscriber);
+      $this->automationEmailScheduler->createSendingTask($newsletter, $subscriber, $this->isTransactional($args));
     } catch (Throwable $e) {
       throw InvalidStateException::create()->withMessage('Could not create sending task.');
     }
   }
 
+  private function isTransactional(StepRunArgs $args): bool {
+    if (!in_array($args->getAutomationRun()->getTriggerKey(), self::TRANSACTIONAL_TRIGGER_KEYS, true)) {
+      return false;
+    }
+
+    //@ToDo: Position 0 = root, Position 1 = triggers, Position 2 is first possible action; validate assumption
+    return $args->getAutomation()->getPositionOfStep($args->getStep()) === 2;
```

Comment:
Linking it with the JS part, that almost works the way I described: https://github.com/mailpoet/mailpoet/pull/4796#discussion_r1151653810

Reply:
> Otherwise, we'd be risking something like this ...

If I understand you correctly, you fear the automation run might have been started with an non-transactional trigger, but we check for this just a few lines up

```
    if (!in_array($args->getAutomationRun()->getTriggerKey(), self::TRANSACTIONAL_TRIGGER_KEYS, true)) {
      return false;
    }
```